packages/sync/src/DataSyncClient.ts

```diff
@@ -0,0 +1,49 @@
+import { DataSyncConfig } from "./config";;
+import { createDefaultLink } from "./links/LinksBuilder";
+
+import { OfflineClient, ApolloOfflineClient } from "offix-client";
+
+/**
+* Factory for creating Apollo Offline Client
+*
+* @param userConfig options object used to build client
+* @deprecated use OfflineClient class directly:
+*  ```javascript
+*  const offlineClient = new OfflineClient(config);
+*  await offlineClient.init();
+*  ```
+*/
+export const createClient = async (userConfig: DataSyncConfig):
+  Promise<DataSyncClient> => {
+  const offlineClient = new DataSyncClient(userConfig);
+  return offlineClient.init();
+};
+
+/**
+ * OfflineClient
+ *
+ * Enables offline workflows, conflict resolution and cache
+ * storage on top Apollo GraphQL JavaScript client.
+ *
+ * Usage:
+ *
+ *  ```javascript
+ *  const offlineClient = new DataSyncClient(config);
+ *  await offlineClient.init();
+ *  ```
+ */
+export class DataSyncClient extends OfflineClient {
+
+  constructor(private userConfig: DataSyncConfig) {
+    super(userConfig);
+  }
+
+  /**
+   * Initialize client
+   */
+  public async init(): Promise<ApolloOfflineClient> {
+    const terminatingLink = await createDefaultLink(this.userConfig);
+    this.config.terminatingLink = terminatingLink;
```

Comment:
That was the reason I wanted init method to get terminatingLink but we can expose config and do it as well.

Reply:
I mean, it would be possible to simply call `createDefaultLink` within `createClient` and add it to the config. Then you wouldn't need to override `init` at all right?

Reply:
Do you think we can return ApolloClient - do you need OfflineClient for anything in your current use cases?